test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -298,6 +309,49 @@ instance (KnownSymbol param, HasPath sub) => HasPath (Capture param t :> sub)
         let (verb, sub) = getPath (Proxy @sub)
         in (verb, "/{" <> symbolVal (Proxy :: Proxy param) <> "}" <> sub)
 
+
+
+{-------------------------------------------------------------------------------
+               roundtripAndGolden for an entire Api type
+-------------------------------------------------------------------------------}
+
+class JsonTestable api where
+    roundtripAndGoldenPerType :: Proxy api -> Spec
```

Comment:
A problem: types appearing multiple times in the Api will also be tested multiple times.

Should return a list or a set of specs instead.

Reply:
Fixed.